met-api/src/met_api/models/engagement.py

```diff
@@ -39,13 +40,34 @@ def get_engagement(cls, engagement_id):
         data = db.session.query(Engagement).filter_by(id=engagement_id).first()
         return engagement_schema.dump(data)
 
+    @classmethod
+    def get_published_engagement(cls, engagement_id):
+        """Get an engagement."""
+        engagement_schema = EngagementSchema()
+        data = db.session.query(Engagement)\
+            .filter(Engagement.status_id != Status.Draft.value)\
+            .filter_by(id=engagement_id)\
+            .first()
+        return engagement_schema.dump(data)
+
     @classmethod
     def get_all_engagements(cls):
         """Get all engagements."""
         engagements_schema = EngagementSchema(many=True)
         data = db.session.query(Engagement).join(EngagementStatus).order_by(Engagement.id.asc()).all()
         return engagements_schema.dump(data)
 
+    @classmethod
+    def get_published_engagements(cls):
+        """Get all open engagements."""
```

Comment:
I generally like get_engagement_status and pass the status 
In the model level , it will be less code for different use cases
but for readability at service , get_published_engagement will be good